src/libPMacc/include/cuSTL/cursor/accessor/LinearInterpAccessor.hpp

```diff
@@ -0,0 +1,167 @@
+ */
+template<typename T_Cursor, int dim = cursor::traits::dim<T_Cursor>::value>
+struct LinearInterpAccessor;
+
+template<typename T_Cursor>
+struct LinearInterpAccessor<T_Cursor, DIM1>
+{
+    typedef T_Cursor Cursor;
+    typedef typename Cursor::ValueType type;
+
+    Cursor cursor;
+
+    /**
+     * @param cursor input data
+     */
+    HDINLINE LinearInterpAccessor(const Cursor& cursor) : cursor(cursor) {}
+
+    template<typename T_Position>
+    HDINLINE type operator()(const T_Position x) const
+    {
+        T_Position intPart;
+        T_Position fracPart;
+
+        fracPart[0] = PMacc::algorithms::math::modf(x[0], &(intPart[0]));
+
+        const math::Int<DIM1> idx1D(static_cast<int>(intPart[0]));
+
+        type result = PMacc::traits::GetInitializedInstance<type>()(0.0);
+        typedef typename T_Position::type PositionComp;
+        for(int i = 0; i < 2; i++)
+        {
+            const PositionComp weighting1D = (i == 0 ? (PositionComp(1.0) - fracPart[0]) : fracPart[0]);
+            result += static_cast<type>(weighting1D * this->cursor[idx1D + math::Int<DIM1>(i)]);
+        }
+
+        return result;
+    }
+};
+
+template<typename T_Cursor>
+struct LinearInterpAccessor<T_Cursor, DIM2>
+{
+    typedef T_Cursor Cursor;
+    typedef typename T_Cursor::ValueType type;
+
+    Cursor cursor;
+
+    /**
+     * @param cursor input data
+     */
+    HDINLINE LinearInterpAccessor(const Cursor& cursor) : cursor(cursor) {}
+
+    template<typename T_Position>
+    HDINLINE type operator()(const T_Position x) const
+    {
+        T_Position intPart;
+        T_Position fracPart;
+
+        fracPart[0] = PMacc::algorithms::math::modf(x[0], &(intPart[0]));
+        fracPart[1] = PMacc::algorithms::math::modf(x[1], &(intPart[1]));
+
+        const math::Int<DIM2> idx2D(static_cast<int>(intPart[0]),
+                                    static_cast<int>(intPart[1]));
+
+        type result = PMacc::traits::GetInitializedInstance<type>()(0.0);
+        typedef typename T_Position::type PositionComp;
+        for(int i = 0; i < 2; i++)
+        {
+            const PositionComp weighting1D = (i == 0 ? (PositionComp(1.0) - fracPart[0]) : fracPart[0]);
+            for(int j = 0; j < 2; j++)
+            {
+                const PositionComp weighting2D = weighting1D * (j == 0 ? (PositionComp(1.0) - fracPart[1]) : fracPart[1]);
+                result += static_cast<type>(weighting2D * this->cursor[idx2D + math::Int<DIM2>(i, j)]);
+            }
+        }
+
+        return result;
+    }
+};
+
+template<typename T_Cursor>
+struct LinearInterpAccessor<T_Cursor, DIM3>
+{
+    typedef T_Cursor Cursor;
+    typedef typename T_Cursor::ValueType type;
+
+    Cursor cursor;
+
+    /**
+     * @param cursor input data
+     */
+    HDINLINE LinearInterpAccessor(const Cursor& cursor) : cursor(cursor) {}
+
+    template<typename T_Position>
+    HDINLINE type operator()(const T_Position x) const
+    {
+        T_Position intPart;
```

Comment:
Please use a static assert to avoid non integer types and check that the dimension of `T_Position` is three.
This avoids out of memory access in line 138-144